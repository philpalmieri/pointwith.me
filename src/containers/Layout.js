import React from 'react';
import PropTypes from 'prop-types';
import { Container, Icon, Image, Menu } from 'semantic-ui-react';

const propTypes = {
  children: PropTypes.node.isRequired,
  contentCenter: PropTypes.bool,
};

const defaultProps = {
  contentCenter: false,
};

const Layout = ({ children, contentCenter }) => {
  return (
    <Container text style={{ marginTim: '1em' }}>
      <Menu fixed="top" inverted>
        <Container text id="topBranding">
          <Image
            size="mini"
            src="/favicon-32x32.png"
            style={{ marginRight: '1.5em' }}
          />
          <p>Planning Poker for Remote Teams!</p>
        </Container>
      </Menu>
      <main className={contentCenter ? 'content-center' : ''}>{children}</main>
      <Container>
        <footer>
          <p>
            <a
              href="https://twitter.com/pointwithme"
              target="_blank"
              rel="noopener noreferrer"
            >
              <Icon name="twitter" />
            </a>
            <a
              href="https://github.com/philpalmieri/pointwith.me"
              target="_blank"
              rel="noopener noreferrer"
            >
              <Icon name="github" />
            </a>
            &copy; 2020 PointWith.me
          </p>
        </footer>
      </Container>
    </Container>
  );
};

Layout.propTypes = propTypes;
Layout.defaultProps = defaultProps;

export default Layout;
