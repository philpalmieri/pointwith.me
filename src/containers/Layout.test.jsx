// Theirs
import React from 'react';
import { render } from '@testing-library/react';

// Ours
import Layout from './Layout';

describe('layout', () => {
  it('should render', () => {
    expect.hasAssertions();

    const { baseElement } = render(
      <Layout>
        <p>Content goes here.</p>
      </Layout>
    );

    expect(baseElement).toHaveTextContent('Content goes here.');
    expect(baseElement).toMatchSnapshot();
  });
});
