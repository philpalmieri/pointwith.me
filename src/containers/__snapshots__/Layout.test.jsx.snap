// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout should render 1`] = `
<body>
  <div>
    <div
      class="ui text container"
    >
      <div
        class="ui inverted top fixed menu"
      >
        <div
          class="ui text container"
          id="topBranding"
        >
          <img
            class="ui mini image"
            src="/favicon-32x32.png"
            style="margin-right: 1.5em;"
          />
          <p>
            Planning Poker for Remote Teams!
          </p>
        </div>
      </div>
      <main
        class=""
      >
        <p>
          Content goes here.
        </p>
      </main>
      <div
        class="ui container"
      >
        <footer>
          <p>
            <a
              href="https://twitter.com/pointwithme"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="twitter icon"
              />
            </a>
            <a
              href="https://github.com/philpalmieri/pointwith.me"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                aria-hidden="true"
                class="github icon"
              />
            </a>
            Â© 2020 PointWith.me
          </p>
        </footer>
      </div>
    </div>
  </div>
</body>
`;
